README.md

```diff
@@ -1,2 +1,359 @@
+          version: ">=5.2"
+          command: rtrace -version  # command to get the version
+          pattern: r'\d+\.\d+'  # regex pattern to extract the version from command output
+```
+
+Here is another one for running a `template` for running `ladybug` Python scripts.
+This `operator` will run on Windows, Linux and Mac platforms. You can exclude a platform
+by removing it from the list.
+
+```yaml
+
+  - name: ladybug-operator
+    container:
+      name: ladybug
+      image: ladybugtools/ladybug:latest
+    local:
+      pip:
+        - name: lbt-ladybug
+          version: ">=0.4.3"
+      language:
+        - name: python
+          version: ">=3.6"
+      platform:
+       - Windows
+       - Linux
+       - Mac
+
+```
+
+You can also import the operator from a `YAML` or `JSON` file using the `import_from`
+key. All the fields except for the name will be loaded from the file. It's good practice
+to use the same name inside the yaml file and the workflow.
+
+```yaml
+
+  - name: ladybug-operator
+    import_from: /operators/ladybug_operator.yaml
+
+```
+
+## 3. templates
+
+Templates are a discrete reusable unit of code that can be executed separately or as part
+of a workflow. Queenbee supports 4 type of objects as templates:
+
+- `module` : A single module. Each module uses an operator from operators section.
+- `dag`: Collection of modules as a Directed Acyclic Graph (DAG).
+- `steps`: Collection of modules to be executed sequentially in steps.
+- `workflow`: A full Queenbee workflow itself can be referenced as a template in another
+  workflow.
+
+Templates can use the `import_from` keyword to import the `template` from a `YAML` or
+`JSON` file. Keep in mind that a template in a file should be self-sufficient and include
+all the information including `operator`s. This is different from when the `template` is
+part of the `workflow` where an `operator` will be referenced by name.
+
+Here is a `module` for generating a sky with desired irradiance which is part of a
+workflow. In order to make it a valid `template` in a separate file you should use the
+full operator specification from the last step instead of only using the name of the
+operator.
+
+```yaml
+  name: generate_sky
+  inputs:
+    parameters:
+      - name: desired_irradiance
+        value: 100000   # default value which can be overwritten
+  operator: radiance-operator
+  # commands and args will be used both locally and inside the container
+  command:
+    - /bin/sh
+    - -c
+  args:
+    - gensky -c -B '{{ inputs.parameters.desired_irradiance }}' > '{{ workflow.artifacts.sky }}'
+  outputs:
+    artifacts:
+      - name: sky
+        path: '{{ workflow.artifacts.sky }}'
+```
+
+Alternatively, you can use `honeybee-radiance` operator which has a command for
+generating sky.
+
+```yaml
+  name: generate_sky
+  inputs:
+    parameters:
+      - name: desired_irradiance
+        value: 100000   # default value which can be overwritten
+  operator: honeybee-radiance
```

Comment:
Where is this `honeybee-radiance` operator and what does it look like?  I can see what the `radiance-operator` looks like in the `operators` section.  At least give me a link to this `honeybee-radiance` operator so that I know what you are talking about here.

Reply:
There is none in this document. I can update the operator for ladybug to be honeybee-radiance operator! I didn't include it as it will be a long one but it makes sense to have one if we are going to reference it.

Reply:
```yaml

  - name: honeybee-radiance
    container:
      name: honeybee-radiance
      image: ladybugtools/honeybee-radiance-workflow:latest
    local:
      app:
        - name: radiance
          version: ">=5.2"
          command: rtrace -version
          pattern: r'\d+\.\d+'
      pip:
        - name: lbt-honeybee
          version: ">=0.4.3"
        - name: honeybee-radiance-workflow
      language:
        - name: python
          version: ">=3.6"
      platform:
       - Windows
       - Linux
       - Mac

```

also added to the document. thanks.